v-next/hardhat/src/internal/builtin-plugins/artifacts/hookHandlers/hre.ts

```diff
@@ -0,0 +1,13 @@
+import type { HardhatRuntimeEnvironmentHooks } from "../../../../types/hooks.js";
+
+import { ArtifactsManagerImplementation } from "../../../artifacts/artifacts-manager.js";
```

Comment:
I'll add this to the guidelines tomorrow, but this is the only  hook handler category where we want to load things lazily, as it's always executed.

Reply:
I switched the `ArtifactsManagerImplementation` to be lazily based on the pattern in the network builtin plugin.